[Student]  Add Project Night Study

tumblecat44 opened the pull request:
## Overview (Required)

-  Create New Night Study Screen
-  Create New Night Study Logic

## Screenshot

| Before | After |
| :--: | :--: |
| <img src="https://github.com/user-attachments/assets/8a79ff62-5455-4dea-b685-3c0547024a2b" width="300" /> | <img src="https://github.com/user-attachments/assets/5a4ad2c8-a671-4551-9601-3ea01e92bf78" width="300" /> |

## Issue

- close #330 
- close #329

kmj5004 committed:
fix: room name

tumblecat44 committed:
Merge remote-tracking branch 'origin/feature/s/329-new-create-nightStudy-logic' into feature/s/330-create-new-night-study-screen

tumblecat44 committed:
feat: Add Ask Project Night Study

tumblecat44 committed:
feat: Create Ask Project Night Study Function

tumblecat44 committed:
feat: Add Ask Project Night Study Form

tumblecat44 committed:
chore: Add Default Place

tumblecat44 committed:
feat: Create Night Study Student Response Model

tumblecat44 committed:
feat: Create Get Night Study Student Feature

tumblecat44 committed:
feat: Create Night Study Student Model

tumblecat44 committed:
feat: Add Method To Get Night Study Students In Repository

tumblecat44 committed:
feat: Add Method To Get Night Study Student In VIewModel

tumblecat44 committed:
chore: Change Property Type From Long To Int

tumblecat44 committed:
chore: Change To Nullable

tumblecat44 committed:
feat: Add Student Search

tumblecat44 committed:
feat: Add Ask Project Night Study

tumblecat44 committed:
feat: Updates Project Name Validation And Submission Logic

Refines the validation and submission process for project night study requests. It adjusts the project name validation to allow any non-empty string, providing a more flexible user experience, and ensures submission is contingent on the project name being present, enhancing data integrity.

tumblecat44 committed:
feat: Removes Unnecessary Key From List

Removes the key parameter from the `items` composable in the
`AskNightStudyScreen`. The key was using the `id` field of the
`filteredStudentList` items, which is no longer necessary for
the list's functionality or performance.

tumblecat44 committed:
chore: Temporarily Comment Out

kmj5004 committed:
feat: project request

tumblecat44 committed:
Merge remote-tracking branch 'origin/feature/s/329-new-create-nightStudy-logic' into feature/s/330-create-new-night-study-screen

tumblecat44 committed:
chore: Replaces getMyBan with getMyProject

Updates the night study screen and view model to fetch project-related data instead of ban data.

This change ensures that the application retrieves the correct
information related to student projects during night study sessions.

tumblecat44 committed:
chore: Apply spotless

tumblecat44 committed:
chore: Apply Spotless

8954sood commented on data/core/src/commonMain/kotlin/com/b1nd/dodam/data/core/model/Place.kt:
```diff
@@ -1,29 +1,48 @@
+    PROJECT6("프로젝트 6실"),
+    LAB1718("랩 17, 18실"),
+    LAB1920("랩 19, 20실"),
+    LAB2122("랩 21, 22"),
```
"랩 21,22실"이 올바른 것 같습니다.

8954sood commented on data/core/src/commonMain/kotlin/com/b1nd/dodam/data/core/model/Place.kt:
```diff
@@ -1,29 +1,48 @@
+    "프로젝트 6실" -> Place.PROJECT6
+    "랩 17, 18실" -> Place.LAB1718
+    "랩 19, 20실" -> Place.LAB1920
+    "랩 21, 22" -> Place.LAB2122
```
위와 동일합니다.

8954sood commented on data/core/src/commonMain/kotlin/com/b1nd/dodam/data/core/model/Place.kt:
```diff
@@ -1,29 +1,48 @@
+    Place.LAB1718 -> "랩 17, 18실"
+    Place.LAB1920 -> "랩 19, 20실"
+    Place.LAB2122 -> "랩 21, 21실"
+}
```
오타가 존재합니다.

8954sood commented on feature-student/ask-nightstudy/src/main/java/com/b1nd/dodam/asknightstudy/AskNightStudyScreen.kt:
```diff
@@ -73,24 +85,52 @@ internal fun AskNightStudyScreen(viewModel: AskNightStudyViewModel = koinViewMod
     var nightStudyEndDate by remember { mutableStateOf(LocalDate.now().plusDays(13)) }
 
-    var nightStudyPlace by remember { mutableStateOf(Place.PROGRAMMING_1) }
+    val projectNightStudyTimeList = arrayListOf("심자 1", "심자 2")
```
해당 변수의 값이 변경되지 않는다면, persistentListOf를 사용하여 불변성을 명시하는 것이 바람직해보입니다.

8954sood commented on feature-student/ask-nightstudy/src/main/java/com/b1nd/dodam/asknightstudy/AskNightStudyScreen.kt:
```diff
@@ -73,24 +85,52 @@ internal fun AskNightStudyScreen(viewModel: AskNightStudyViewModel = koinViewMod
+    val nightTypeList = listOf(
+        "개인",
+        "프로젝트",
+    )
```
위와 같이 persistentListOf를 사용하는 것이 바람직해보입니다.

8954sood commented on feature-student/ask-nightstudy/src/main/java/com/b1nd/dodam/asknightstudy/AskNightStudyScreen.kt:
```diff
@@ -253,45 +326,96 @@ internal fun AskNightStudyScreen(viewModel: AskNightStudyViewModel = koinViewMod
+                            if (projectNightStudyReason.length !in 1..250) "프로젝트 이름을 입력해주세요" else ""
+                        } else {
+                            if (nightStudyReason.length !in 10..250) "사유를 10자 이상 입력해주세요." else ""
+                        },
```
아래와 같이 가독성을 개선하는 것이 좋아보입니다.

```kotlin
supportText = when {
    nightTypeIndex.isProject() && projectNightStudyReason.length !in 1..250 ->
        "프로젝트 이름을 입력해주세요"
    !nightTypeIndex.isProject() && nightStudyReason.length !in 10..250 ->
        "사유를 10자 이상 입력해주세요."
    else -> ""
}
```

tumblecat44 replied:
수정하겠습니다

8954sood commented on feature-student/nightstudy/src/main/java/com/b1nd/dodam/nightstudy/NightStudyScreen.kt:
```diff
@@ -168,205 +197,400 @@ fun NightStudyScreen(
+                        when (val nightStudyUiState = uiState) {
+                            is NightStudyUiState.Success -> {
+                                if (nightStudyUiState.nightStudies.isNotEmpty()) {
+                                    Log.d("nighStudy", "NightStudyScreen: Is")
```
디버깅 용 로그는 지워주세요

8954sood commented on network/night-study/src/commonMain/kotlin/com/b1nd/dodam/network/nightstudy/model/NightStudyRequest.kt:
```diff
@@ -13,3 +13,15 @@ NightStudyRequest(
+
+@Serializable
+data class
+ProjectRequest(
```
들여쓰기가 잘못된 것 같습니다.

tumblecat44 committed:
chore: Updates place name

Updates place name to be more descriptive.

tumblecat44 committed:
chore: Fixes typo in Place enum

Corrects a typo in the "Place" enum to ensure consistency
and proper mapping of location names.

tumblecat44 committed:
chore: Fixes typo in project place mapping

Corrects a typo in the mapping of Place.PROJECT6, ensuring accurate representation of "프로젝트 6실".

tumblecat44 committed:
chore: Uses immutable list for night study times

Changes the `projectNightStudyTimeList` from a mutable `ArrayList` to an immutable `PersistentList`.

This improves performance by avoiding unnecessary object mutations.

tumblecat44 committed:
chore: Refactors conditional logic for support text

Improves readability and maintainability by using a `when` statement to handle different support text messages based on night study type and input length.

tumblecat44 committed:
chore: Removes unnecessary log statement

Removes an unnecessary log statement from the NightStudyScreen.

The log statement was likely used for debugging purposes and is no longer needed in the production code.

tumblecat44 committed:
chore: Removes unnecessary line breaks

Removes unnecessary line breaks within data class definitions for improved code readability.

8954sood reviewed (approved):
LGTM 👍
